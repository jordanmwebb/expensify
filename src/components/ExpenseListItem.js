import React from 'react';
import { Link } from 'react-router-dom';


const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`} >Edit Expense</Link>
        <h3>{description}</h3>
        <p>{amount} - {createdAt}</p>
    </div>
);

export default ExpenseListItem;