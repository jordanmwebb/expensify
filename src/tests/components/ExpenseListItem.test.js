import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render an expense list item', () => {
    const wrapper = shallow(<ExpenseListItem id={expenses[0].id} 
                                            description={expenses[0].description} 
                                            amount={expenses[0].amount}
                                            note={expenses[0].note}
                                            createdAt={expenses[0].createdAt} />);
    expect(wrapper).toMatchSnapshot();
});