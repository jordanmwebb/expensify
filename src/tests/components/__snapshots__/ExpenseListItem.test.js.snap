// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an expense list item 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    Edit Expense
  </Link>
  <h3>
    Gum
  </h3>
  <p>
    $1.95
    -
    December 31st, 1969
  </p>
</div>
`;
